### PR TITLE
BPSO-24495

### PR DESCRIPTION
**Pull request by sushant1245**
#PATCH#

# CHANGELOG

* I did performance enhancement for select-input component
* I have modified 'didrecieveAttrs' & 'hasQueryChanged' method for select-input component 
* I improved performance by 50% depending on length of bunsen form

### CODE REVIEW

File: addon/components/select-input.js [hunk 1]
```diff
@@ -50,21 +50,24 @@ export default AbstractInput.extend({
...
-    const value = this.get('bunsenStore.formValue')
-    const bunsenId = this.get('bunsenId')
-
-    if (this.hasQueryChanged(oldAttrs, newAttrs, modelDef.query) &&
-      utils.hasValidQueryValues(value, modelDef.query, bunsenId)) {
-      listUtils.getOptions(value, modelDef, bunsenId, dbStore).then((opts) => {
-        this.set('options', opts)
-      })
+   if (this.hasQueryChanged(oldAttrs, newAttrs, modelDef.query)) {
+           //setting required variables once above condition is true
```
Comment (job13er) [inline addon/components/select-input.js]: Looks like you're indenting is all off, which probably has something to do with the build failing.

Comment (job13er) [inline addon/components/select-input.js]: I'd run `npm run lint` and even `npm test` locally before submitting a PR, it'll save time and show you exactly what needs attention.

**Comment by job13er** [comment]
I see changes to source files, and no changes to test files... That's generally not a good sign. If the change to source files is an additional feature, there should be new tests to cover that feature. If the change to source files is a bugfix, there should be a test that was added which fails without the bugfix. In this case, for the bugfix, there should be a test added which fails without the presence of the fix and passes with the fix.

**Comment by job13er** [comment]
You are also missing a scope in your PR description, so that `pr-bumper` can know how to bump the version when the PR is merged. We should really add a `CONTRIBUTING.md` and `PULL_REQUEST_TEMPLATE.md` to this repo to introduce `pr-bumper` to potential contributors...

In the meantime, take a look [here](https://github.com/ciena-blueplanet/pr-bumper#pull-requests) for more information.

**Comment by sglanzer** [comment]
Can you update the title of this PR to reference the functionality fixed/added instead of a JIRA number?  The numbers aren't meaningful in a commit log

**Comment by sushant1245** [comment]
There is already a test case for select-input named as "frost-bunsen-input-select-test.js" , we have only optimize the performance of didReceiveAttrs method of select-input & no change in existing functionality of component. Hence not writing any new test case.

File: addon/components/select-input.js [hunk 2]
```diff
@@ -50,20 +50,23 @@ export default AbstractInput.extend({
...
-    const value = this.get('bunsenStore.formValue')
-    const bunsenId = this.get('bunsenId')
-
-    if (this.hasQueryChanged(oldAttrs, newAttrs, modelDef.query) &&
-      utils.hasValidQueryValues(value, modelDef.query, bunsenId)) {
-      listUtils.getOptions(value, modelDef, bunsenId, dbStore).then((opts) => {
-        this.set('options', opts)
-      })
+    if (this.hasQueryChanged(oldAttrs, newAttrs, modelDef.query)) {
+           // setting required variables once above condition is true
```
Comment (sandersky) [inline addon/components/select-input.js]: Spacing is off.

File: addon/components/select-input.js [hunk 3]
```diff
@@ -75,36 +78,53 @@ export default AbstractInput.extend({
    * @returns {Boolean} true if query has been changed
    */
   hasQueryChanged (oldAttrs, newAttrs, modelQuery) {
-    // allow models that don't have query defined to pass as well as
-    // allow the options to get initially populated
+     // allow models that don't have query defined to pass as well as
+     // allow the options to get initially populated
```
Comment (sandersky) [inline addon/components/select-input.js]: Added spaces are unnecessary.

File: addon/components/select-input.js (hunk 2 above)
Comment (sushant1245) [inline addon/components/select-input.js]: Are you referring to to the extra space @ line 59 
"// setting required variables once above condition is true"

**Comment by psbanka** [comment]
approve

**Pull request merged**